pkg/controller/compute/v1beta1/cluster.go

```diff
@@ -0,0 +1,287 @@
+/*
+Copyright 2019 The Crossplane Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"strings"
+	"time"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/pkg/errors"
+	container "google.golang.org/api/container/v1beta1"
+	"google.golang.org/api/option"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	"github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	runtimev1alpha1 "github.com/crossplaneio/crossplane-runtime/apis/core/v1alpha1"
+	"github.com/crossplaneio/crossplane-runtime/pkg/logging"
+	"github.com/crossplaneio/crossplane-runtime/pkg/meta"
+	"github.com/crossplaneio/crossplane-runtime/pkg/resource"
+
+	"github.com/crossplaneio/stack-gcp/apis/compute/v1beta1"
+	gcpv1alpha3 "github.com/crossplaneio/stack-gcp/apis/v1alpha3"
+	gcp "github.com/crossplaneio/stack-gcp/pkg/clients"
+	gke "github.com/crossplaneio/stack-gcp/pkg/clients/container"
+)
+
+const (
+	controllerName    = "gke.compute.gcp.crossplane.io"
+	finalizer         = "finalizer." + controllerName
+	clusterNamePrefix = "gke-"
+
+	requeueOnWait   = 30 * time.Second
+	requeueOnSucces = 2 * time.Minute
+
+	updateErrorMessageFormat         = "failed to update cluster object: %s"
+	erroredClusterErrorMessageFormat = "gke cluster is in %s state with message: %s"
```

Comment:
We can probably write uppercase GKE here. Usually the rule is that error strings aren't capitalised, but acronyms get an exception.